java/src/shared/model/bank/ResourceCardBank.java

```diff
@@ -52,14 +88,264 @@ public ResourceCardBank (Player player) {
+        if (ownedByGame) {
+            throw new Exception("Must specify Resource Type to draw from Game");
+        } else {
+            List<ResourceCard> hand = new ArrayList<>();
+            for (ResourceCard brick : bricks) {
+                hand.add(brick);
+            }
+            for (ResourceCard wood : woods) {
+                hand.add(wood);
+            }
+            for (ResourceCard sheep : sheeps) {
+                hand.add(sheep);
+            }
+            for (ResourceCard wheat : wheats) {
+                hand.add(wheat);
+            }
+            for (ResourceCard ore : ores) {
+                hand.add(ore);
+            }
+            return hand.get(new Random().nextInt(hand.size()));
+        }
+    }
+
+    public int getNumberOfBrick() {
+        return bricks.size();
+    }
+
+    public int getNumberOfOre() {
+        return ores.size();
+    }
+
+    public int getNumberOfSheep() {
+        return sheeps.size();
+    }
+
+    public int getNumberOfWheat() {
+        return wheats.size();
+    }
+
+    public int getNumberOfWood() {
+        return woods.size();
     }
 
     /**
      * @return number of cards in the ResourceCardBank
      */
+    @Override
     public int size() {
-        return resourceCards.size();
+        return getNumberOfBrick() + getNumberOfOre() + getNumberOfSheep() + getNumberOfWheat() + getNumberOfWood();
+    }
+
+    @Override
+    public ResourceCard discard(ResourceType type) throws InsufficientResourcesException, InvalidTypeException {
+        switch (type) {
+            case BRICK:
+                if (getNumberOfBrick() > 0) {
+                    return bricks.remove(0);
+                } else {
+                    throw new InsufficientResourcesException("There are no bricks to discard");
+                }
+            case ORE:
+                if (getNumberOfOre() > 0) {
+                    return ores.remove(0);
+                } else {
+                    throw new InsufficientResourcesException("There are no ores to discard");
+                }
+            case SHEEP:
+                if (getNumberOfSheep() > 0) {
+                    return sheeps.remove(0);
+                } else {
+                    throw new InsufficientResourcesException("There are no sheep to discard");
+                }
+            case WHEAT:
+                if (getNumberOfWheat() > 0) {
+                    return wheats.remove(0);
+                } else {
+                    throw new InsufficientResourcesException("There are no wheats to discard");
+                }
+            case WOOD:
+                if (getNumberOfWood() > 0) {
+                    return woods.remove(0);
+                } else {
+                    throw new InsufficientResourcesException("There are no woods to discard");
+                }
+            default:
+                throw new InvalidTypeException("The given type is invalid");
+        }
```

Comment:
This switch-case is very similar to what is in the `draw` function. Maybe we should pull it out into it's own `removeResource` function?